dask_searchcv/adaptive.py

```diff
@@ -0,0 +1,385 @@
+            ``pd.DataFrame(self.cv_results_)``. Each dictionary contains the
+            keys ``test_score``, ``params`` and ``param_{p}`` for each ``p``
+            in params. That is, ``params['alpha'] == params_alpha``.
+
+        best_params_ : dict
+            The best parameters found by this model selection algorithm. This
+            is determined by cross validation.
+
+        best_estimator_ : type(model)
+            The best estimator as determined by this model selection algorithm.
+            This is determined by cross validation.
+
+        best_index_ : int
+            The index of the best performing estimator and parameters in
+            ``best_estimator_`` and ``best_params_``
+
+        history : list of dicts
+            The recorded history for this adaptive process. It includes the
+            evaluations on each step, and it's possible to use this to see each
+            decision on which models to keep evaluating. Each dictionary in this
+            variables has ``set(keys) == {}``
+
+        Notes
+        -----
+        This implements the `Hyperband model selection algorithm`_
+        and only requires one input, ``max_iter`` which is porprotional to
+        "computational effort" required. Notably it does not require balancing
+        a trade-off between the number of params to evaluate and how long to
+        train each config.
+
+        Hyperband has some theoritical guarantees on finding (close to) the
+        optimal config from ``params`` given this computational effort
+        ``max_iter``.
+
+        .. _Hyperband model selection algorithm: https://arxiv.org/abs/1603.06560
+        """
+        self.params = params
+        self.model = model
+        self.R = max_iter
+        self.eta = eta
+        self.best_val_score = -np.inf
+        self.n_iter = 0
+        self.run_in_parallel = run_in_parallel
+
+        if not hasattr(model, 'warm_start'):
+            warnings.warn('model has no attribute warm_start. Hyperband will assume it '
+                          'is reusing previous calls to `partial_fit` during each call '
+                          'to `partial_fit`')
+        else:
+            if not model.warm_start:
+                raise ValueError('Hyperband relies on model(..., warm_start=True).')
+
+        self.history = []
+
+        super_default = dict(scoring=None, iid=True, refit=True, cv=None,
+                             error_score='raise', return_train_score=_RETURN_TRAIN_SCORE_DEFAULT,
+                             scheduler=None, n_jobs=-1, cache_cv=True)
+        for k, v in super_default.items():
+            if k in kwargs and kwargs[k] != super_default[k]:
+                warnings.warn('Hyperband ignores the keyword argument {k}.')
+        super(Hyperband, self).__init__(model, **kwargs)
+
+    def fit(self, X, y, dry_run=False, verbose=False, **fit_kwargs):
+        """
+        This function implicitly assumes that higher scores are better.
```

Comment:
These docs would be better to have in the class docstring, since they're describing the parameters like `model`, which were passed above.

Also, you can safely assume a few things, since they're part of the scikit-learn API

- higher scores are better
- the signature of `model.set_params`
- `sklearn.base.clone` will work.

It's good to call out `partial_fit`.

Reply:
I added some words on why I included these lines in the docs. I see this as being useful for other classes that do not conform to `sklearn.BaseEstimator` classes.

I've raised a ValueError if the estimator doesn't have `partial_fit`, included a helpful error message and tested for it.